src/crewai/agents/agent_interface.py

```diff
@@ -0,0 +1,52 @@
+from abc import ABC, abstractmethod
+from typing import Optional, List, Any, Dict
+
+from pydantic import BaseModel, PrivateAttr, Field
+
+from crewai.utilities import I18N
+
+
+class AgentWrapperParent(ABC, BaseModel):
```

Comment:
Really loved the generic agent interface. In a perfect world, this would be the only way the engine would refer to agents, right!? Should we call it `AgentInterface`, like the file name? #namingishard

Reply:
Just to make it clear, the dream here is that even `Agent` respects `AgentInterface`, then anything that complies to the interface would be able to be an agent.

Reply:
Good suggestion,  that was exactly the idea :)